src/liballoc/sync.rs

```diff
@@ -2279,13 +2292,6 @@ unsafe fn data_offset<T: ?Sized>(ptr: *const T) -> isize {
     unsafe { data_offset_align(align_of_val(&*ptr)) }
```

Comment:
*oops* this line should have been `align_of_val_raw` now, right?

But I doubt that that's enough to fix https://github.com/rust-lang/rust/issues/80365.

Reply:
Yes, it definitely should have been. I don't know how I missed this one 🙃